asax/calculator.py

```diff
@@ -29,46 +29,34 @@ def update(self, atoms: Atoms):
             raise RuntimeError("Need an Atoms object to do anything!")
 
         if self.atoms is None:
-            self.atoms = atoms.copy()
+            # new or reset calculator: redo everything
+            self.on_atoms_changed(atoms)
+            self.atoms = atoms.copy()  # avoid missing changes
             self.results = {}
-            self.on_atoms_changed()
             self.setup()
             return
```

Comment:
I'd put everything from line 32 to 37 into a dedicated method `reinitialize()` or something. IMHO this all is more confusing than it should be. `update()` should only check for changes in `atoms` and delegate accordingly. 

Setting `self.atoms = None` and recursively calling `self.update(atoms)` is essentially a complicated way to get to line 32 (what could be a simple call of `reinitialize()`).

Reply:
Yes, that is correct. I'll do this. `update` in general is a bad name, but I can't think of anything better.

Reply:
Ok, made a commit in that direction.